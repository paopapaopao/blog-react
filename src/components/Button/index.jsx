import PropTypes from "prop-types";
import React from "react";

const Button = ({
  className,
  color,
  disabled,
  onClick,
  text,
  type,
  variant,
  ...rest
}) => {
  let colorClass = "";
  switch (color) {
    case "secondary":
      colorClass = "secondary";
      break;
    case "success":
      colorClass = "success";
      break;
    case "danger":
      colorClass = "danger";
      break;
    case "warning":
      colorClass = "warning";
      break;
    case "info":
      colorClass = "info";
      break;
    case "light":
      colorClass = "light";
      break;
    case "dark":
      colorClass = "dark";
      break;
    default:
      colorClass = "primary";
      break;
  }

  let variantClass = "";
  switch (variant) {
    case "outlined":
      variantClass = "btn-outline";
      break;
    default:
      variantClass = "btn";
      break;
  }

  const buttonClassName = `btn ${variantClass}-${colorClass} ${className}`;

  return (
    <button
      className={buttonClassName}
      disabled={disabled}
      onClick={onClick}
      type={type}
      {...rest}
    >
      {text}
    </button>
  );
};

Button.propTypes = {
  className: PropTypes.string,
  color: PropTypes.oneOf([
    "danger",
    "dark",
    "info",
    "light",
    "primary",
    "secondary",
    "success",
    "warning"
  ]),
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
  text: PropTypes.string.isRequired,
  type: PropTypes.oneOf(["button", "reset", "submit"]),
  variant: PropTypes.oneOf(["contained", "outlined"])
};

Button.defaultProps = {
  className: "",
  color: "primary",
  disabled: false,
  onClick: () => {},
  type: "button",
  variant: "contained"
};

export default Button;
