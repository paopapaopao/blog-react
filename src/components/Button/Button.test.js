import React from "react";
import { render } from "@testing-library/react";
import Button from ".";

it("renders an abled, contained, primary (default) button correctly", () => {
  const { asFragment } = render(<Button text="Button" />);

  expect(asFragment()).toMatchSnapshot();
});

it("renders a disabled button correctly", () => {
  const { asFragment } = render(<Button disabled text="Button" />);

  expect(asFragment()).toMatchSnapshot();
});

it("renders an outlined button correctly", () => {
  const { asFragment } = render(<Button text="Button" variant="outlined" />);

  expect(asFragment()).toMatchSnapshot();
});

it("renders a secondary button correctly", () => {
  const { asFragment } = render(<Button color="secondary" text="Button" />);

  expect(asFragment()).toMatchSnapshot();
});
