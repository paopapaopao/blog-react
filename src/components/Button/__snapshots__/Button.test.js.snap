// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a disabled button correctly 1`] = `
<DocumentFragment>
  <button
    class="btn btn-primary "
    disabled=""
    type="button"
  >
    Button
  </button>
</DocumentFragment>
`;

exports[`renders a secondary button correctly 1`] = `
<DocumentFragment>
  <button
    class="btn btn-secondary "
    type="button"
  >
    Button
  </button>
</DocumentFragment>
`;

exports[`renders an abled, contained, primary (default) button correctly 1`] = `
<DocumentFragment>
  <button
    class="btn btn-primary "
    type="button"
  >
    Button
  </button>
</DocumentFragment>
`;

exports[`renders an outlined button correctly 1`] = `
<DocumentFragment>
  <button
    class="btn btn-outline-primary "
    type="button"
  >
    Button
  </button>
</DocumentFragment>
`;
